runtime/lua/vim/_helptags.lua

```diff
@@ -0,0 +1,106 @@
+local M = {}
+
+local ts = vim.treesitter
+
+local query = ts.query.parse('vimdoc', [[
+    (tag (word) @tagname)
+]])
+
+--- Find all helptags in a single file.
+--- @param filename string helpfile with tags
+--- @return { [1]: string, [2]: string, [3]: string}[] tuple of tag, file, and search command
+local function extract_file_tags(filename)
+  local file = assert(io.open(filename, 'r'))
+  local source = file:read('*a')
+  file:close()
```

Comment:
Is `assert()` fine here or should this report a Vim-style error to the user (E40)? Or ignore?